apps/fly-my-rockets/src/app/rockets/services/photo-upload/photo-upload.service.ts

```diff
@@ -21,4 +22,20 @@ export class PhotoUploadService {
       })
     );
   }
+
+  getMetadata(ref: string): Observable<any> {
+    // Using AngularFireStorage#ref().getMetadata wasn't returning. So
+    // use raw storage API instead.
+    const storage = firebase.storage();
+    const getMetadata = storage.ref(ref).getMetadata();
+    return timer(1_000).pipe(
+      switchMap(() => {
+        return from(getMetadata).pipe(
+          catchError((err) => {
+            return this.getMetadata(ref);
+          })
```

Comment:
This is used to see if a thumbnail is available yet. Wait 1 second initially before checking. If it errors then the thumbnail hasn't been generated yet so loop on it until it's finished processing.

Probably should be capping this at some point so we don't get into an infinite loop should something with processing go wrong. Though the user can always hit Cancel on the dialog and that should unsubscribe from everything.